### ISSUE TITLE
Escape `;` character in tracking segments

### ISSUE THREAD

**Issue by OliverJAsh** [unsplash/imgix-trackable#5]
Tracking segments could include a `;`, in which case we would need to escape them.

Perhaps we should URI encode each `;` separated segment, before base 64 encoding the whole thing?

@TimmyCarbone @lukechesser

**Comment by lukechesser** [comment]
Can you think of an example of where `;` would come up? I guess from search queries or app name where the user has control over the input?

Makes sense to me

**Comment by OliverJAsh** [comment]
Yeah, that sort of thing! Property and label could be any string.

**Comment by lukechesser** [comment]
Cool, I'm down. @TimmyCarbone does that work for you?

It may be easiest for Tim to also use this module for decoding on the data repo. Thoughts?

**Comment by TimmyCarbone** [comment]
Makes sense to me yes.
I prefer to do the escaping in BigQuery to keep the load as low as possible on the dyno, but yeah ... I could use the module as well to be more consistent and avoid breaking changes.

**Comment by lukechesser** [comment]
@TimmyCarbone oh yeah fair. Could you do the escaping BigQuery side and then still use this module on your side? Or would that double escape?

**Comment by TimmyCarbone** [comment]
I don't have a use for the module if I escape in BigQuery :)
The module would replace all the treatment that I'm currently doing in BigQuery.

It's either BigQuery processing or node processing with the module.

**Comment by lukechesser** [comment]
Ah okay. Nvm then @TimmyCarbone. Like you said, better to use the power of BigQuery than process it inside JS. I guess just look into the code here if you ever notice a discrepancy 👍

**Comment by lukechesser** [comment]
@OliverJAsh So to be clear, we still want to escape characters when encoding and unescape when decoding in this module.
